common/meta/types/proto/meta.proto

```diff
@@ -87,10 +83,72 @@ message WatchResponse {
   Event event = 1;
 }
 
+// messages for txn
+message TxnCondition {
+  // compare target enum
+  enum ConditionTarget {
+    Key = 0;
+    Value = 1;
+  }
+
+  // condition result
+  enum ConditionResult {
+    // only used for key compare
+    Exist = 0;
+    NotExist = 1;
+
+    // only used for value compare
+    Equal = 2;
+    Greater = 3;
+    Less = 4;
+    NotEqual = 5;    
+  }
+
+  TxnKey key = 1;
+  ConditionTarget target = 2;
+
+  optional SeqV value = 3;
+
+  // the expected result of condition, if `expected` match the condition result,
+  // then the `if_then` op will be executed, otherwise `else_then` op will be executed.
+  ConditionResult expected = 4;
+}
+
+message TxnOp {
+  oneof request {
+    GetRequest get = 1;
+    PutRequest put = 2;
+    DeleteRequest delete = 3;
+  }
+}
+
+message TxnOpResponse {
+  oneof response {
+    GetResponse get = 1;
+    PutResponse put = 2;
+    DeleteResponse delete = 3;
+  }
+}
+
+message TxnRequest {
+  TxnCondition cond = 1;
```

Comment:
add a `repeated` for conditions?

i remembers that it's allowed to pass multiple conditions in an atomic manner in the etcd api

Reply:
ok